templates/boltzmann/middleware/honeycomb.ts

```diff
@@ -9,77 +10,151 @@ import beeline from 'honeycomb-beeline'
+          activeContext = otelPropagation.extract(activeContext, {
+            traceparent: context.headers.traceparent,
+            tracestate: context.headers.tracestate
+          })
+        }
+
+        const rootSpan = tracer.startSpan(`${context.method} ${context.url.pathname}${context.url.search}`, {
+          attributes: {
+            'request.host': context.host,
+            'request.original_url': context.url.href,
+            'request.remote_addr': context.remote,
+            'request.method': context.method,
+            'request.scheme': context.url.protocol,
+            'request.path': context.url.pathname,
+            'request.query': context.url.search
+          }
+        }, activeContext)
+
+        otelTrace.setSpan(activeContext, rootSpan)
+
+        // do not do as I do,
+        onHeaders(context._response, function () {
+          rootSpan.setAttribute('response.status_code', String(context._response.statusCode))
+          if (context.handler) {
+            const handler = context.handler;
+            if (handler.route) {
+              rootSpan.setAttribute('request.route', handler.route)
+            }
+            if (handler.method) {
+              rootSpan.setAttribute('request.method', handler.method)
+            }
+            if (handler.version) {
+              rootSpan.setAttribute('request.version', handler.version)
+            }
+          }
+          Object.entries(context.params).map(([key, value]) => {
+            rootSpan.setAttribute(`request.param.${key}`, value)
+          })
+
+          rootSpan.end()
+        })
 
-      if (!trace) {
         return next(context)
       }
-
-      const boundFinisher = beeline.bindFunctionToTrace((response: ServerResponse) => {
-        beeline.addContext({
-          'response.status_code': String(response.statusCode)
-        })
-
-        beeline.addContext({
-          'request.route': context.handler.route,
-          'request.method': context.handler.method,
-          'request.version': context.handler.version
+    }
+  } else {
+    const schema = require('honeycomb-beeline/lib/schema')
+    const tracker = require('honeycomb-beeline/lib/async_tracker')
+
+    return function honeycombTrace (next: Handler) {
+      return (context: Context) => {
+        const traceContext = _getBeelineTraceContext(context)
+        const trace = beeline.startTrace({
+          [schema.EVENT_TYPE]: 'boltzmann',
+          [schema.PACKAGE_VERSION]: '1.0.0',
+          [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
+          [schema.TRACE_ID_SOURCE]: traceContext.source,
+          'request.host': context.host,
+          'request.original_url': context.url.href,
+          'request.remote_addr': context.remote,
+          'request.method': context.method,
+          'request.scheme': context.url.protocol,
+          'request.path': context.url.pathname,
+          'request.query': context.url.search
```

Comment:
We should use the constants exported from the 
`const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions');`
`SemanticResourceAttributes.HOST` (not exact name)
etc

Reply:
I made this change for otel. Do we want to change this for beelines too? My instincts say no but I'm open to being swayed.